### PR TITLE
Feature/136 modify auto field using variable name

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Fix number operation and logical operation
- [x] Modify fnString to handle formula with(out) `function` wrapper
- [x] Modify fnString to replace variable name with question id
- [x] https://github.com/akvo/akvo-react-form/issues/138

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206597446546951

### CODE REVIEW

File: src/fields/TypeMultipleOption.jsx [hunk 1]
```diff
@@ -66,7 +68,11 @@ const TypeMultipleOption = ({
   }, [currentValue, updateDataPointName]);
 
   useEffect(() => {
-    setOptions([...option, ...extraOption]);
+    const _options = [...option, ...extraOption].map((o) => ({
```
Comment (dedenbangkit) [inline src/fields/TypeMultipleOption.jsx:71]: ...option.map, ...extraOption

File: src/fields/TypeMultipleOption.jsx [hunk 2]
```diff
@@ -40,6 +41,7 @@ const TypeMultipleOption = ({
     ? extra.filter((ex) => ex.placement === 'after')
     : [];
   const currentValue = form.getFieldValue([id]);
+  const labelText = label || name;
```
Comment (dedenbangkit) [inline src/fields/TypeMultipleOption.jsx:44]: remove labelText

File: src/fields/TypeInput.jsx [hunk 3]
```diff
@@ -28,6 +29,7 @@ const TypeInput = ({
     ? extra.filter((ex) => ex.placement === 'after')
     : [];
   const currentValue = form.getFieldValue([id]);
+  const labelText = label || name;
```
Comment (dedenbangkit) [inline src/fields/TypeInput.jsx:32]: remove labelText

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
remove labelText and use label || name directly
```bash
src/fields/TypeText.jsx
24:  const labelText = label || name;
32:          content={labelText}

src/fields/TypeOption.jsx
57:  const labelText = label || name;
120:          content={labelText}

src/fields/TypeTable.jsx
23:  const labelText = label || name;
51:            content={labelText}

src/fields/TypeCascade.jsx
248:  const labelText = label || name;
303:        name={labelText}
327:          content={labelText}

src/fields/TypeTree.jsx
64:  const labelText = label || name;
71:          content={labelText}

src/fields/TypeNumber.jsx
42:  const labelText = label || name;
87:          content={labelText}

src/fields/TypeMultipleOption.jsx
44:  const labelText = label || name;
88:          content={labelText}

src/fields/TypeDate.jsx
29:  const labelText = label || name;
65:          content={labelText}

src/fields/TypeImage.jsx
62:  const labelText = label || name;
89:            content={labelText}

src/fields/TypeAutoField.jsx
226:  const labelText = label || name;
243:          content={labelText}

src/fields/TypeInput.jsx
32:  const labelText = label || name;
63:          content={labelText}

src/fields/TypeGeo.jsx
27:  const labelText = label || name;
36:            content={labelText}
```

**Comment by ifirmawan** [comment]
> remove labelText and use label || name directly
> 
> ```shell
> src/fields/TypeText.jsx
> 24:  const labelText = label || name;
> 32:          content={labelText}
> 
> src/fields/TypeOption.jsx
> 57:  const labelText = label || name;
> 120:          content={labelText}
> 
> src/fields/TypeTable.jsx
> 23:  const labelText = label || name;
> 51:            content={labelText}
> 
> src/fields/TypeCascade.jsx
> 248:  const labelText = label || name;
> 303:        name={labelText}
> 327:          content={labelText}
> 
> src/fields/TypeTree.jsx
> 64:  const labelText = label || name;
> 71:          content={labelText}
> 
> src/fields/TypeNumber.jsx
> 42:  const labelText = label || name;
> 87:          content={labelText}
> 
> src/fields/TypeMultipleOption.jsx
> 44:  const labelText = label || name;
> 88:          content={labelText}
> 
> src/fields/TypeDate.jsx
> 29:  const labelText = label || name;
> 65:          content={labelText}
> 
> src/fields/TypeImage.jsx
> 62:  const labelText = label || name;
> 89:            content={labelText}
> 
> src/fields/TypeAutoField.jsx
> 226:  const labelText = label || name;
> 243:          content={labelText}
> 
> src/fields/TypeInput.jsx
> 32:  const labelText = label || name;
> 63:          content={labelText}
> 
> src/fields/TypeGeo.jsx
> 27:  const labelText = label || name;
> 36:            content={labelText}
> ```

Done

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**